lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -170,1648 +169,171 @@ std::vector<const char *> MakeArgv(const llvm::ArrayRef<std::string> &strs) {
+    // Searching backward so that we choose the variable in closest scope
+    // among variables of the same name.
+    for (int64_t i = end_idx - 1; i >= 0; --i) {
+      lldb::SBValue curr_variable = top_scope->GetValueAtIndex(i);
+      std::string variable_name = CreateUniqueVariableNameForDisplay(
+          curr_variable, is_duplicated_variable_name);
+      if (variable_name == name) {
+        variable = curr_variable;
+        break;
+      }
     }
-  }
-  llvm::json::Array targets;
-
-  bool had_escape_prefix =
-      llvm::StringRef(text).starts_with(dap.command_escape_prefix);
-  ReplMode completion_mode = dap.DetectReplMode(frame, text, true);
-
-  // Handle the offset change introduced by stripping out the
-  // `command_escape_prefix`.
-  if (had_escape_prefix) {
-    if (offset < static_cast<int64_t>(dap.command_escape_prefix.size())) {
-      body.try_emplace("targets", std::move(targets));
-      response.try_emplace("body", std::move(body));
-      dap.SendJSON(llvm::json::Value(std::move(response)));
-      return;
+  } else {
+    // This is not under the globals or locals scope, so there are no duplicated
+    // names.
+
+    // We have a named item within an actual variable so we need to find it
+    // withing the container variable by name.
+    lldb::SBValue container = dap.variables.GetVariable(variablesReference);
+    variable = container.GetChildMemberWithName(name.data());
+    if (!variable.IsValid()) {
+      if (name.starts_with("[")) {
+        llvm::StringRef index_str(name.drop_front(1));
+        uint64_t index = 0;
+        if (!index_str.consumeInteger(0, index)) {
+          if (index_str == "]")
+            variable = container.GetChildAtIndex(index);
+        }
+      }
     }
-    offset -= dap.command_escape_prefix.size();
   }
+  return variable;
+}
 
-  // While the user is typing then we likely have an incomplete input and cannot
-  // reliably determine the precise intent (command vs variable), try completing
-  // the text as both a command and variable expression, if applicable.
-  const std::string expr_prefix = "expression -- ";
-  std::array<std::tuple<ReplMode, std::string, uint64_t>, 2> exprs = {
-      {std::make_tuple(ReplMode::Command, text, offset),
-       std::make_tuple(ReplMode::Variable, expr_prefix + text,
-                       offset + expr_prefix.size())}};
-  for (const auto &[mode, line, cursor] : exprs) {
-    if (completion_mode != ReplMode::Auto && completion_mode != mode)
-      continue;
-
-    lldb::SBStringList matches;
-    lldb::SBStringList descriptions;
-    if (!dap.debugger.GetCommandInterpreter().HandleCompletionWithDescriptions(
-            line.c_str(), cursor, 0, 100, matches, descriptions))
-      continue;
+// Both attach and launch take a either a sourcePath or sourceMap
+// argument (or neither), from which we need to set the target.source-map.
+void SetSourceMapFromArguments(DAP &dap, const llvm::json::Object &arguments) {
```

Comment:
Is this repeated? Or I think maybe this is still here for the existing `request_launch` function that hasn't moved over yet.

Should this move to the a RequestHelpers.h like we have for EventHelpers.h?

Reply:
Yeah, I managed to avoid dit by migrating to LaunchRequestHandler & RestartRequestHandler.